test/with_dummyserver/test_https.py

```diff
@@ -471,16 +471,22 @@ def setUp(self):
         self._pool = HTTPSConnectionPool(self.host, self.port)
 
     def test_set_ssl_version_to_sslv3(self):
-        self._pool.ssl_version = ssl.PROTOCOL_SSLv3
-        self.assertRaises(SSLError, self._pool.request, 'GET', '/')
+        self.assertEqual(
+            ssl.PROTOCOL_SSLv3,
+            util.ssl_.resolve_ssl_version(ssl.PROTOCOL_SSLv3)
+        )
 
     def test_ssl_version_as_string(self):
-        self._pool.ssl_version = 'PROTOCOL_SSLv3'
-        self.assertRaises(SSLError, self._pool.request, 'GET', '/')
+        self.assertEqual(
+            ssl.PROTOCOL_SSLv3,
+            util.ssl_.resolve_ssl_version("PROTOCOL_SSLv3")
+        )
 
     def test_ssl_version_as_short_string(self):
-        self._pool.ssl_version = 'SSLv3'
-        self.assertRaises(SSLError, self._pool.request, 'GET', '/')
+        self.assertEqual(
+            ssl.PROTOCOL_SSLv3,
+            util.ssl_.resolve_ssl_version("SSLv3")
+        )
```

Comment:
As best as I could see, this is actually the only bit of functionality that the tests were executing.

As to "do we want to get rid of it", I think we have to: they rely on the notion that SSLv3 is actually present in the OpenSSL on the system, and that's frequently untrue.